### PR TITLE
fix: Fix dataserver REST endpoints error and improvement documentation (PTL-1204)

### PR DESCRIPTION
**Pull request by jacinpoz**
…

Thank you for contributing to the documentation.

Do the changes you have made apply to both Current and Previous versions?
No

Have you checked all new or changed links?
Yes
Is there anything else you would like us to know?

The rest of the rest endpoints documentation should probably be updated to follow the same structure.

For reference: 

  - We have an [internal contributions guide](https://www.notion.so/genesisglobal/Contributing-new-documentation-75953fb245f246ff872789035451a0c4)
  - We have a [style guide](https://www.notion.so/genesisglobal/Documentation-style-guide-5b04ec6fe12f4262b90d192effd8059b) 

**This week's exciting excerpts from the style guide**

- We write in UK English, with Oxford English Dictionary (OED) spellings, grammar and vocabulary.  

Use the present tense wherever possible. Only use another tense where it is strictly necessary.

- Present tense (preferred whenever possible): *The script creates a new folder.*
- Future tense (avoid whenever possible): *The script will create a new folder.*

### CODE REVIEW

File: docs/03_server/11_integration/01_rest-endpoints/03_basics.mdx [hunk 1]
```diff
@@ -79,232 +79,498 @@ Data Servers are the most complex of the REST Endpoints provided by the Genesis
...
 
-There follows a list of Data Server messages mapped to HTTP endpoints.
+### Endpoint
+The naming approach for the dataserver endpoint follows the lowercase hyphen-separated web standards convention (i.e. ALL_TRADES becomes all-trades). There is no need to rename your dataserver resource, the endpoint will be exposed using the correct naming convention automatically.
 
-### DATA_LOGON
-
-To initiate a Data Server query, a DATA_LOGON message is required to create the subscription. This is requested via a POST request to
+A sample endpoint url would look like this:
 `[host]:[genesis_router_port]/[data_server_query_name]`.
```
Comment (jacinpoz) [inline docs/03_server/11_integration/01_rest-endpoints/03_basics.mdx:86]: we repeat this URL over and over for every single dataserver REST endpoint description. I have created a new Endpoint heading at the beginning to explain what the URL should look like.

File: docs/03_server/11_integration/01_rest-endpoints/03_basics.mdx [hunk 2]
```diff
@@ -79,232 +79,498 @@ Data Servers are the most complex of the REST Endpoints provided by the Genesis
 
 For all Data Server requests, a `SOURCE_REF` header is mandatory, as it is used to match requests to the correct client subscription. This in turn allows for multiple subscriptions to the same query for the same session (think multiple grids in the UI using the same query with different filters).
 
-There follows a list of Data Server messages mapped to HTTP endpoints.
+### Endpoint
+The naming approach for the dataserver endpoint follows the lowercase hyphen-separated web standards convention (i.e. ALL_TRADES becomes all-trades). There is no need to rename your dataserver resource, the endpoint will be exposed using the correct naming convention automatically.
 
-### DATA_LOGON
```
Comment (jacinpoz) [inline docs/03_server/11_integration/01_rest-endpoints/03_basics.mdx:84]: I have divided the different headings into REST methods: POST, GET, PUT and DELETE. Given this is documentation about rest endpoints it feels more natural to define them that way.

Comment (wjhendry) [reply docs/03_server/11_integration/01_rest-endpoints/03_basics.mdx:84]: that makes sense.

File: docs/03_server/11_integration/01_rest-endpoints/03_basics.mdx (hunk 1 above)
Comment (wjhendry) [reply docs/03_server/11_integration/01_rest-endpoints/03_basics.mdx:86]: Good idea

**Review by wjhendry — APPROVED** [review]

**Pull request merged**